app/assets/javascripts/discourse/app/lib/textarea-text-manipulation.js

```diff
@@ -746,7 +752,31 @@ export default class TextareaTextManipulation {
     this.$textarea.attr("dir", newDir).focus();
   }
 
-  autocomplete() {
-    return this.$textarea.autocomplete(...arguments);
+  autocomplete(options) {
+    return this.$textarea.autocomplete(
+      options instanceof Object
+        ? { textManipulation: this, ...options }
+        : options
+    );
+  }
+
+  replaceTerm({ start, end, term }) {
+    let text = this.value;
+    const space = text.substring(end + 1, end + 2) === " " ? "" : " ";
+    text =
+      text.substring(0, start) +
+      term +
+      space +
+      text.substring(end + 1, text.length);
+    this.value = text;
+    setCaretPosition(this.textarea, start + 1 + term.trim().length);
+  }
+
+  getCaretPosition() {
+    return caretPosition(this.textarea);
+  }
+
+  getCaretCoords(start) {
+    return this.$textarea.caretPosition({ pos: start + 1 });
```

Comment:
can't we avoid using the jquery object here?

Reply:
It's because we're using the jQuery plugin: https://github.com/discourse/discourse/blob/603600166761ba77d5bb2f5224e9a7683a1ab6ee/app/assets/javascripts/discourse/app/lib/caret-position.js

Reply:
ah yes forgot about this lib, yes sure no need to block on this